### PR TITLE
Reader: Enable photon on external images with size parameters

### PR DESCRIPTION
**Pull request by akirk**
#### Changes proposed in this Pull Request

Before this patch, if an image contains any parameter, we just don't display it since we can't route it through Photon. Often images, though, already contain some "size parameters"  (like `w`), either because it already uses a Photon site that we cannot detect (like an Atomic site) or some other internal service to resize the image. 

Part of the problem is also that we attach URL parameters to URLs that we think might be Photon URLs, causing the non-action down the stream (since the URL now has URL parameters).

With this patch, we attempt to remove the size URL parameters and also only modify known Photon URLs.

#### Testing instructions

Find a site through a search that has image URLs (see p5PDj3-4Up-p2) and verify that the images now show.

### CODE REVIEW

File: client/lib/safe-image-url/index.js [hunk 1]
```diff
@@ -51,22 +58,34 @@ export default function safeImageUrl( url ) {
 		return url;
 	}
 
-	const { hostname, pathname, search } = getUrlParts( url );
+	const parsedUrl = getUrlParts( url );
+	if ( ! parsedUrl.protocol ) {
+		return url;
```
Comment (bluefuton) [inline client/lib/safe-image-url/index.js:63]: Is the intention here to allow relative URLs through? Could this potentially miss a protocol-relative image URL like `//mybadsite.com/badimage.jpg`?

**Comment by bluefuton** [comment]
Thanks for taking this on @akirk! 🏅 

Can we add a few tests to verify this behaviour? There are already some for `makeImagesSafe` that we could add to:

https://github.com/Automattic/wp-calypso/blob/46be9e8721255c2da4a53d709a8c27d50cd75bf3/client/lib/post-normalizer/test/index.js#L396

File: client/lib/safe-image-url/index.js (hunk 1 above)
Comment (akirk) [reply client/lib/safe-image-url/index.js]: I've removed it.

**Comment by akirk** [comment]
I've added unit tests, could you take another look? Thanks!

**Review by bluefuton — APPROVED** [review]
🚢 Looks good to me. Thanks for fixing this!

**Pull request merged**